ext/Errno/Errno_pm.PL

```diff
@@ -183,16 +176,16 @@ sub get_files {
 		if (/$pat/o) {
 		   my $f = $1;
 		   $f =~ s,\\\\,/,g;
-		   $file{$f} = 1;
+		   push(@file, $f);
 		}
 	    }
 	    else {
-		$file{$1} = 1 if /$pat/o;
+		push(@file, $1) if /$pat/o;
 	    }
 	}
 	close(CPPO);
     }
-    return keys %file;
+    return @file;
```

Comment:
This adds duplicate entries to get_files() which really isn't necessary.

Rather than using an array I was thinking of something more like:
```
        my $seq  = 1;
  	while(<CPPO>) {
	    if ($^O eq 'os2' or $IsMSWin32) {
		if (/$pat/o) {
		   my $f = $1;
		   $f =~ s,\\\\,/,g;
		   $file{$f} = $seq++ if !$file{$f};
		}
	    }
	    else {
		$file{$1} = $seq++ if /$pat/o && !$file{$f};
	    }
	}
	close(CPPO);
    }
    return sort { $file{$a} <=> $file{$b} } keys %file;
```

This returns the filenames in the order seen and avoids returning duplicate entries.

Reply:
Yes, I was aware, and initially tried to use List::Util::uniq on the return; alas, this is not functioning at that point (because it being a miniperl IIUC). So I figured it didn't truly matter, including the same file many times shouldn't cause a problem.

But yes, I checked and there's a fair amount of unnecessary duplication, so I made a local uniq() function to be used for the return list. I think this a bit clearer than having the %seen hash in the main part, and since it's always a list, it will also work for any of the other OS/variant cases in case they have more than one entry in the array.

Pushed a rebased/squashed commit with all changes (also getting rid of some commits I made from work which uses a different mail address in the commit, causing the porting/author.t to complain).

?